libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -429,14 +426,69 @@ public boolean checkBirthDateCorrelationInCredentials(String userId)
                 <= 1;
     }
 
+    // This method checks the birthdate requirement, which is in violation of GPG45 rules.
+    // However, considering that BAV and Address CRI  does not contain a birthdate,
+    // we add this special handling only for BAV and Address CRI. For other CRIs, we continue to
+    // validate this
+    // requirement.
+    private List<IdentityClaim> getIdentityClaimsForBirthDateCorrelation(
+            List<VcStoreItem> vcStoreItems) throws HttpResponseExceptionWithErrorBody {
+        List<IdentityClaim> identityClaims = new ArrayList<>();
+        for (VcStoreItem item : vcStoreItems) {
+            IdentityClaim identityClaim =
+                    getIdentityClaim(item.getCredential(), item.getCredentialIssuer(), true);
+            if (isBirthDateEmpty(identityClaim.getBirthDate())) {
+                // excluded BAV and Address CRI
+                if (EXCLUDE_CRIS_TYPES_FOR_DOB_CORRELATION.contains(item.getCredentialIssuer())) {
+                    continue;
+                }
+                addLogMessage(item, "Birthdate property is missing from VC");
+            }
+            identityClaims.add(identityClaim);
+        }
+        return identityClaims;
+    }
+
+    private boolean isBirthDateEmpty(List<BirthDate> birtDates) {
+        return CollectionUtils.isEmpty(birtDates)
+                || birtDates.stream().map(BirthDate::getValue).allMatch(StringUtils::isEmpty);
```

Comment:
I don't think you need the empty check? - `allMatch` should return true if the list is empty

Reply:
If birthDates list comes as null., a NullPointerException will be thrown. I think, we should keep it.